### ISSUE TITLE
Request full support for Vite (eg reimplement Babel plugin in Vite too)

### ISSUE THREAD

**Issue by lloydjatkinson** [styled-components/styled-components#3759]
If styled-components could have an official Vite plugin I feel that the React ecosystem overall would benefit greatly. There is no reason the existing babel plugin couldn't exist alongside it.

The motivation for this is that Vite is increasingly becoming the single build tool that more or less replaces N number of previous tools. Additionally, given the recent discussions in the ecosystem about not using CRA due to it's technical debt and reliance on webpack with the a popular consensus being to use Vite for new React apps, it's only a matter of time before Vite + styled-components becomes a popular requested feature.

I've created this ticket as a starting point for the discussion.

**Comment by quantizor** [comment]
Closing — this issue is stale (2+ years without resolution) and relates to an older version of styled-components. If this is still reproducible on the latest v6.x release, please open a new issue with a minimal reproduction.

**Comment by lloydjatkinson** [comment]
Agreed. I recommend https://vanilla-extract.style/ for a framework-independent solution.
